sdk/samples/iot/sample_pnp_thermostat_component.h

```diff
@@ -9,6 +9,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include <azure/core/az_json.h>
```

Comment:
Lol, why does this keep getting added? I am fine with it, but just curious :)

Reply:
It's really, the compilation fails without it. I don't know why id doesn't always. Include file rearrangement has something to do with it? Include file rearrangement should not normally break compilation, as the inclusion should take care of including everything it needs. And if you look in this file, it uses json types, but nowhere in the chain of include in this file does it include json.h. So it must be that the files using this header just always happened to have json.h included before. So, this is an improvement.

Reply:
Agreed.

And I think you are right. The include reordering is probably breaking the build in some cases.

Reply:
I will look at the re-ording when I get to this sample for finalizing.  Thanks for finding this!